plan/pattern.go

```diff
@@ -20,8 +20,18 @@ type Pattern interface {
 //  /   \
 // A     B
 func Pat(kind ProcedureKind, predecessors ...Pattern) Pattern {
-	return &OneKindPattern{
-		kind:         kind,
+	return &UnionKindPattern{
+		kinds:        []ProcedureKind{kind},
+		predecessors: predecessors,
+	}
+}
+
+// Union pat matches any plan node from a given set of ProcedureKind and whose
+// predecessors match the given predecessor patterns. This is idencal to Pat,
+// except for matching any pattern root from a set of ProcedureKinds.
+func UnionPat(kinds []ProcedureKind, predecessors ...Pattern) Pattern {
```

Comment:
Nitpick: I like the name `OneOf` better. So I would prefer something like:

```
func OneOf(kinds ...ProcedureKind) Pattern
```

I also don't think you need `predecessors` here. You can do:

```
plan.Pat(universe.FilterKind, plan.OneOf(universe.MaxKind, universe.MinKind), universe.WindowKind)
```

This way we don't have to expect that all patterns will be recursive. There's one pattern for a pattern chain and the other patterns are for specific nodes.

Reply:
The `max`/`min` need to be at the root so it has to go the other way around.

Reply:
Using `OneOf` was actually my first thought because I noticed that elsewhere in the code, but I didn't want it to look too similar to `One` as the struct was already named `OneKindPattern`. Seemed like `OneOfKindPattern` might get missed as something distinct. But yeah, I can definitely change it to keep language consistent.

Reply:
Do you think it would be better to update the `plan.Pat` function to take in a pattern?

There are a lot of ways this can go, but yours might be the best way to change it in a minimal way. I'm going to click approve. I do like the name `OneOf` for a function name, but we can keep `UnionKindPattern` as I really like that name for the type itself.

Reply:
It does though, no?

Reply:
What I meant was as the first parameter. Right now it takes a `ProcedureKind`.

I was just musing to myself. I think this is a good change to the planner that is iterative and useful.

Reply:
Oh I see, yeah I'd have to think about that more.